sklearn/decomposition/_nmf.py

```diff
@@ -1304,24 +1246,97 @@ def fit_transform(self, X, y=None, W=None, H=None):
         W : ndarray of shape (n_samples, n_components)
             Transformed data.
         """
-        X = self._validate_data(X, accept_sparse=('csr', 'csc'),
-                                dtype=[np.float64, np.float32])
+        n_components = self.n_components
+        init = self.init
+        solver = self.solver
+        beta_loss = self.beta_loss
+        tol = self.tol
+        max_iter = self.max_iter
+        random_state = self.random_state
+        alpha = self.alpha
+        l1_ratio = self.l1_ratio
+        verbose = self.verbose
+        shuffle = self.shuffle
+        regularization = self.regularization
+        update_H = self.update_H
```

Comment:
This is unecessary, let's just use the attributes everywhere

Reply:
`n_components` is managed differently though: the  `test_estimators[NMF()-check_estimators_overwrite_params]` fails otherwise.

Reply:
Yes, the ones that we want to modify require to pass through an auxilliary variable.